GGP fails to build a Docker container when running inside of Docker

This should be detected and prevented until there is a fix. Stack trace:

```
[ERROR] BasicDeploymentHelper: Container build failed
[ERROR] AwsGreengrassProvisioner: java.io.IOException: request to write '8' bytes exceeds size in header of '0' bytes for entry 'proc/fs/xfs/xqm'
java.lang.RuntimeException: java.io.IOException: request to write '8' bytes exceeds size in header of '0' bytes for entry 'proc/fs/xfs/xqm'
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicDeploymentHelper.execute(BasicDeploymentHelper.java:750)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicDeploymentHelper.execute(BasicDeploymentHelper.java:62)
        at com.awslabs.aws.greengrass.provisioner.interfaces.helpers.Operation.executeOrDisplayHelp(Operation.java:33)
        at com.awslabs.aws.greengrass.provisioner.interfaces.helpers.Operation.lambda$execute$1(Operation.java:21)
        at io.vavr.control.Try.onSuccess(Try.java:712)
        at com.awslabs.aws.greengrass.provisioner.interfaces.helpers.Operation.execute(Operation.java:21)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$null$2(AwsGreengrassProvisioner.java:86)
        at java.util.Optional.map(Optional.java:215)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$run$da6d192b$1(AwsGreengrassProvisioner.java:86)
        at io.vavr.control.Try.of(Try.java:75)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.run(AwsGreengrassProvisioner.java:81)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.runProvisioner(AwsGreengrassProvisioner.java:59)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.lambda$main$249538b4$1(AwsGreengrassProvisioner.java:31)
        at io.vavr.control.Try.of(Try.java:75)
        at com.awslabs.aws.greengrass.provisioner.AwsGreengrassProvisioner.main(AwsGreengrassProvisioner.java:31)
Caused by: java.io.IOException: request to write '8' bytes exceeds size in header of '0' bytes for entry 'proc/fs/xfs/xqm'
        at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.write(TarArchiveOutputStream.java:449)
        at java.nio.file.Files.copy(Files.java:2909)
        at java.nio.file.Files.copy(Files.java:3069)
        at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:303)
        at com.spotify.docker.client.CompressedDirectory$Visitor.visitFile(CompressedDirectory.java:253)
        at java.nio.file.Files.walkFileTree(Files.java:2670)
        at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:115)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1492)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1465)
        at com.spotify.docker.client.DefaultDockerClient.build(DefaultDockerClient.java:1458)
        at com.awslabs.aws.greengrass.provisioner.implementations.helpers.BasicDeploymentHelper.execute(BasicDeploymentHelper.java:742)
        ... 14 more
        Suppressed: java.io.IOException: This archive contains unclosed entries.
                at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.finish(TarArchiveOutputStream.java:289)
                at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.close(TarArchiveOutputStream.java:307)
                at com.spotify.docker.client.CompressedDirectory.create(CompressedDirectory.java:120)
                ... 18 more
[ERROR] AwsGreengrassProvisioner: java.io.IOException: request to write '8' bytes exceeds size in header of '0' bytes for entry 'proc/fs/xfs/xqm'
```

--- a/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicDeploymentArgumentHelper.java
+++ b/src/main/java/com/awslabs/aws/greengrass/provisioner/implementations/helpers/BasicDeploymentArgumentHelper.java
@@ -108,6 +108,11 @@ public class BasicDeploymentArgumentHelper implements DeploymentArgumentHelper {
         deploymentArguments.s3Bucket = getValueOrDefault(deploymentArguments.s3Bucket, getStringDefault(defaults, "conf.s3Bucket"));
         deploymentArguments.s3Directory = getValueOrDefault(deploymentArguments.s3Directory, getStringDefault(defaults, "conf.s3Directory"));
 
+        if (deploymentArguments.pushContainer) {
+            // If they want to push a container then we have to build it
+            deploymentArguments.buildContainer = true;
+        }
+
         if (deploymentArguments.ec2Launch && deploymentArguments.dockerLaunch && (deploymentArguments.launch != null)) {
             throw new RuntimeException("The EC2 launch, Docker launch, and launch options are mutually exclusive.  Only specify one of them.");
         }
@@ -135,14 +140,13 @@ public class BasicDeploymentArgumentHelper implements DeploymentArgumentHelper {
             deploymentArguments.scriptOutput = true;
         }
 
-        if (deploymentArguments.pushContainer) {
-            // If they want to push a container then we have to build it
-            deploymentArguments.buildContainer = true;
-        }
-
         if ((deploymentArguments.dockerLaunch) || (deploymentArguments.buildContainer)) {
             // Force OEM file output with Docker launch or container build
             deploymentArguments.oemOutput = true;
+        }
+
+        if ((deploymentArguments.buildContainer) && (ioHelper.isRunningInDocker())) {
+            throw new RuntimeException("Can't build a container from inside of Docker yet");
         }
 
         if ((deploymentArguments.launch != null)) {

